Mathlib/Algebra/Category/ModuleCat/Basic.lean

```diff
@@ -42,22 +42,22 @@ open CategoryTheory.Limits.WalkingParallelPair
 
 universe v u
 
-variable (R : Type u) [Ring R]
+variable (R : Type u) [Semiring R]
 
 /-- The category of R-modules and their morphisms.
 
-Note that in the case of `R = ℤ`, we cannot
-impose here that the `ℤ`-multiplication field from the module structure is defeq to the one coming
-from the `isAddCommGroup` structure (contrary to what we do for all module structures in
+Note that in the case of `R = ℕ`, we cannot
+impose here that the `ℕ`-multiplication field from the module structure is defeq to the one coming
+from the `isAddCommMonoid` structure (contrary to what we do for all module structures in
 mathlib), which creates some difficulties down the road. -/
 structure ModuleCat where
   private mk ::
   /-- the underlying type of an object in `ModuleCat R` -/
   carrier : Type v
-  [isAddCommGroup : AddCommGroup carrier]
+  [isAddCommMonoid : AddCommMonoid carrier]
```

Comment:
In case `R` is a ring, this does not change the category of modules up to an equivalence, but if `M : ModuleCat R` and `m : M`, we would no longer be able to write an equality such as `m - m = 0`. This seems to be an excessively radical change to me. Here, we may relax that `Ring` assumption as `Semiring`, but it would want modules over semirings without `neg`, I think you need to define a separate category.